src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -827,7 +827,7 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
 
     case ThreadNetworkDiagnostics::Attributes::RoutingRole::Id: {
         otDeviceRole role = otThreadGetDeviceRole(mOTInst);
-        err               = encoder.Encode(role);
+        err               = encoder.Encode(static_cast<uint8_t>(role));
```

Comment:
What is the actual type of `role`?  Why is this not lossy?

Reply:
The type of `role` is `otDeviceRole` which is  a enumeration type in the range of 0 to 4. It will be cast to `int` type and cause the compilation error `ambiguous reference` for ESP32C3 if we don't do the static cast.